torch/csrc/inductor/static_cuda_launcher.cpp

```diff
@@ -52,8 +54,14 @@ const at::cuda::NVRTC& nvrtc() {
 
 CUdeviceptr getPointer(PyObject* obj) {
   CUdeviceptr data_ptr = 0;
+
   if (THPUtils_checkLong(obj)) {
+#if defined(USE_ROCM)
+    data_ptr = reinterpret_cast<hipDeviceptr_t>(THPUtils_unpackUInt64(obj));
```

Comment:
Why this is needed? And why not do the `reinterpret_cast<CUdeviceptr>` unconditionally?